pkg/flowaggregator/flowaggregator_test.go

```diff
@@ -34,8 +35,129 @@ const (
 	testObservationDomainID = 0xabcd
 )
 
-// TODO: We will add another test for sendDataRecord when we support adding multiple records to single set.
-// Currently, we are supporting adding only one record from one flow key to the set.
+// TODO: Currently, we are supporting adding only one record from one flow key to the set. We will improve TestFlowAggregator_sendFlowKeyRecord when we support adding multiple records to single set.
+
+func TestFlowAggregator_sendFlowKeyRecord(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockIPFIXExpProc := ipfixtest.NewMockIPFIXExportingProcess(ctrl)
+	mockIPFIXRegistry := ipfixtest.NewMockIPFIXRegistry(ctrl)
+	mockDataSet := ipfixtest.NewMockIPFIXSet(ctrl)
+	mockRecord := ipfixtest.NewMockIPFIXRecord(ctrl)
+	mockAggregationProcess := ipfixtest.NewMockIPFIXAggregationProcess(ctrl)
+
+	fa := &flowAggregator{
+		"",
+		"",
+		"tcp",
+		nil,
+		mockAggregationProcess,
+		testExportInterval,
+		mockIPFIXExpProc,
+		testTemplateIDv4,
+		testTemplateIDv6,
+		mockIPFIXRegistry,
+		mockDataSet,
+		"",
+		nil,
+		testObservationDomainID,
+	}
+
+	testcases := []struct {
+		name       string
+		isIPv6     bool
+		flowKey    ipfixintermediate.FlowKey
+		flowRecord ipfixintermediate.AggregationFlowRecord
+	}{
+		{
+			"IPv4_ready_to_send",
+			false,
+			ipfixintermediate.FlowKey{
+				SourceAddress:      "10.0.0.1",
+				DestinationAddress: "10.0.0.2",
+				Protocol:           6,
+				SourcePort:         1234,
+				DestinationPort:    5678,
+			},
```

Comment:
Define the keys once and use them here to avoid repetition.

Reply:
Fixed, thanks!